WordPress/WordPressShare/ShareViewController.swift

```diff
@@ -0,0 +1,156 @@
+        }
+    }
+
+    override func configurationItems() -> [AnyObject]! {
+        let blogPickerItem = SLComposeSheetConfigurationItem()
+        blogPickerItem.title = NSLocalizedString("Post to:", comment: "Upload post to the selected Site")
+        blogPickerItem.value = selectedSiteName ?? NSLocalizedString("Select a site", comment: "Select a site in the share extension")
+        blogPickerItem.tapHandler = { [weak self] in
+            self?.displayBlogPicker()
+        }
+        
+        let statusPickerItem = SLComposeSheetConfigurationItem()
+        statusPickerItem.title = NSLocalizedString("Post Status:", comment: "Post status picker title in Share Extension")
+        statusPickerItem.value = self.postStatuses[postStatus]!
+        statusPickerItem.tapHandler = { [weak self] in
+            self?.displayStatusPicker()
+        }
+        
+        return [blogPickerItem, statusPickerItem]
+    }
+
+    
+    private func displayBlogPicker() {
+        let pickerViewController = BlogPickerViewController()
+        pickerViewController.onChange = { (siteId, description) in
+            self.selectedSiteID = siteId
+            self.selectedSiteName = description
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    private func displayStatusPicker() {
+        let pickerViewController = PostStatusPickerViewController()
+        pickerViewController.statuses = postStatuses
+        pickerViewController.onChange = { (status, description) in
+            self.postStatus = status
+            self.reloadConfigurationItems()
+        }
+        
+        pushConfigurationViewController(pickerViewController)
+    }
+    
+    
+    // TODO: This should eventually be moved into WordPressComKit
+    private let postStatuses = [
+        "draft" : NSLocalizedString("Draft", comment: "Draft post status"),
+        "publish" : NSLocalizedString("Publish", comment: "Publish post status")]
+
+    private func splitContentTextIntoSubjectAndBody(contentText: String) -> (subject: String, body: String) {
+        let fullText = contentText
+        let firstCarriageReturnIndex = fullText.rangeOfCharacterFromSet(NSCharacterSet.newlineCharacterSet())
+        let firstLineOfText = firstCarriageReturnIndex != nil ? fullText.substringToIndex(firstCarriageReturnIndex!.startIndex) : fullText
+        let restOfText = firstCarriageReturnIndex != nil ? fullText.substringFromIndex(firstCarriageReturnIndex!.endIndex) : ""
+
+        return (firstLineOfText, restOfText)
+    }
+    
+    private func contentWithSourceURL(url: NSURL?) -> String {
+        guard let url = url else {
+            return contentText
+        }
+        
+        // Append the URL to the content itself
+        return contentText + "\n\n<a href=\"\(url.absoluteString)\">\(url.absoluteString)</a>"
```

Comment:
Been digging into this. As far as i can tell, the `extensionContext` gives us access to the **public.url** and **attributedContext**, but the website's title isn't there.

We could probably run a NSURLRequest and fetch it, but we're probably better off lightweight, for the first iteration.


```
(lldb) po extensionContext!.inputItems
▿ 1 elements
  - [0] : <NSExtensionItem: 0x7ffc99419410> - userInfo: {
    NSExtensionItemAttachmentsKey =     (
        "<NSItemProvider: 0x7ffc99419d30> {types = (\n    \"public.url\"\n)}"
    );
    NSExtensionItemAttributedContentTextKey = <7b5c7274 66315c61 6e73695c 616e7369 63706731 3235320a 7b5c666f 6e747462 6c5c6630 5c667377 6973735c 66636861 72736574 30204865 6c766574 6963613b 7d0a7b5c 636f6c6f 7274626c 3b5c7265 64323535 5c677265 656e3235 355c626c 75653235 353b7d0a 5c706172 645c7478 3536305c 74783131 32305c74 78313638 305c7478 32323430 5c747832 3830305c 74783333 36305c74 78333932 305c7478 34343830 5c747835 3034305c 74783536 30305c74 78363136 305c7478 36373230 5c706172 6469726e 61747572 616c5c70 61727469 67687465 6e666163 746f7230 0a0a5c66 305c6673 3234205c 63663020 476f6f67 6c657d>;
    supportsJavaScript = 1;
}
```